internal/integrationtest/board/board_test.go

```diff
@@ -0,0 +1,838 @@
+			"archive_filename": "arduinoOTA-1.2.1-linux_386.tar.bz2",
+			"url": "http://downloads.arduino.cc/tools/arduinoOTA-1.2.1-linux_386.tar.bz2",
+			"size": 2133779
+		  },
+		  {
+			"checksum": "SHA-256:5b82310d53688480f34a916aac31cd8f2dd2be65dd8fa6c2445262262e1948f9",
+			"host": "x86_64-linux-gnu",
+			"archive_filename": "arduinoOTA-1.2.1-linux_amd64.tar.bz2",
+			"url": "http://downloads.arduino.cc/tools/arduinoOTA-1.2.1-linux_amd64.tar.bz2",
+			"size": 2257689
+		  },
+		  {
+			"checksum": "SHA-256:ad54b3dcd586212941fd992bab573b53d13207a419a3f2981c970a085ae0e9e0",
+			"host": "arm-linux-gnueabihf",
+			"archive_filename": "arduinoOTA-1.2.1-linux_arm.tar.bz2",
+			"url": "http://downloads.arduino.cc/tools/arduinoOTA-1.2.1-linux_arm.tar.bz2",
+			"size": 2093132
+		  },
+		  {
+			"checksum": "SHA-256:ad54b3dcd586212941fd992bab573b53d13207a419a3f2981c970a085ae0e9e0",
+			"host": "aarch64-linux-gnu",
+			"archive_filename": "arduinoOTA-1.2.1-linux_arm.tar.bz2",
+			"url": "http://downloads.arduino.cc/tools/arduinoOTA-1.2.1-linux_arm.tar.bz2",
+			"size": 2093132
+		  },
+		  {
+			"checksum": "SHA-256:93a6d9f9c0c765d237be1665bf7a0a8e2b0b6d2a8531eae92db807f5515088a7",
+			"host": "i386-apple-darwin11",
+			"archive_filename": "arduinoOTA-1.2.1-darwin_amd64.tar.bz2",
+			"url": "http://downloads.arduino.cc/tools/arduinoOTA-1.2.1-darwin_amd64.tar.bz2",
+			"size": 2244088
+		  },
+		  {
+			"checksum": "SHA-256:e1ebf21f2c073fce25c09548c656da90d4ef6c078401ec6f323e0c58335115e5",
+			"host": "i686-mingw32",
+			"archive_filename": "arduinoOTA-1.2.1-windows_386.zip",
+			"url": "http://downloads.arduino.cc/tools/arduinoOTA-1.2.1-windows_386.zip",
+			"size": 2237511
+		  }
+		]
+	  }
+	],
+	"identification_properties": [
+	  {
+		"properties": {
+		  "vid": "0x2341",
+		  "pid": "0x8057"
+		}
+	  },
+	  {
+		"properties": {
+		  "vid": "0x2341",
+		  "pid": "0x0057"
+		}
+	  }
+	],
+	"programmers": [
+	  {
+		"platform": "Arduino SAMD Boards (32-bits ARM Cortex-M0+)",
+		"id": "edbg",
+		"name": "Atmel EDBG"
+	  },
+	  {
+		"platform": "Arduino SAMD Boards (32-bits ARM Cortex-M0+)",
+		"id": "atmel_ice",
+		"name": "Atmel-ICE"
+	  },
+	  {
+		"platform": "Arduino SAMD Boards (32-bits ARM Cortex-M0+)",
+		"id": "sam_ice",
+		"name": "Atmel SAM-ICE"
+	  }
+	]
+  }
+` // noqa: E501
+
+func TestBoardList(t *testing.T) {
+	if os.Getenv("CI") != "" {
+		t.Skip("VMs have no serial ports")
+	}
+
+	env, cli := integrationtest.CreateArduinoCLIWithEnvironment(t)
+	defer env.CleanUp()
+
+	_, _, err := cli.Run("core", "update-index")
+	require.NoError(t, err)
+	stdout, _, err := cli.Run("board", "list", "--format", "json")
+	require.NoError(t, err)
+	// check is a valid json and contains a list of ports
+	var ports []string
+	err = json.Unmarshal(stdout, &ports)
+	require.NoError(t, err)
+	require.NotEmpty(t, ports)
+	for _, port := range ports {
+		require.Contains(t, port, "protocol")
+		require.Contains(t, port, "protocol_label")
+	}
+}
```

Comment:
As I explained in the single [commit](https://github.com/arduino/arduino-cli/pull/1847/commits/6b88371585d644b0498d23f77a015ecfb3ad7a0e), I could not test this function properly because I did not have any boards to test it with, so `board list` always returned an empty json like `[ ]`. I chose to unmarshal into a list of strings because the original test was making the check `assert isinstance(ports, list)`, but I could not be sure of it.